google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/DoManagedVmDeployment.java

```diff
@@ -117,13 +142,59 @@ public void undeploy(@NotNull UndeploymentTaskCallback callback) {
             }
           });
         } else {
-          callback.errorOccurred("Deployment failed with exit code: " + event.getExitCode());
+          callback.errorOccurred(
+              GctBundle.message("appengine.deployment.error", event.getExitCode()));
+        }
+        if (appDefaultCredentialsPath.exists()) {
+          appDefaultCredentialsPath.delete();
         }
       }
     });
     processHandler.startNotify();
   }
 
+  private static final String CLIENT_ID_LABEL = "client_id";
+  private static final String CLIENT_SECRET_LABEL = "client_secret";
+  private static final String REFRESH_TOKEN_LABEL = "refresh_token";
+  private static final String GCLOUD_USER_TYPE_LABEL = "type";
+  private static final String GCLOUD_USER_TYPE = "authorized_user";
```

Comment:
I typically do that for class fields as well as constants that figure prominently in a class or are exported. In this case because they are so loca, I'm tempted to leave here or possible even in-line.. What do you think?